docs/source/amp.rst

```diff
@@ -0,0 +1,31 @@
+.. role:: hidden
+    :class: hidden-section
+
+Automatic Mixed Precision package - torch.cuda.amp
+==================================================
+
+.. automodule:: torch.cuda.amp
+.. currentmodule:: torch.cuda.amp
+
+``torch.cuda.amp`` provides convenience methods for running networks with mixed precision,
+where some operations use the ``torch.float32`` (``float``) datatype and other operations
+use ``torch.float16`` (``half``). Some operations, like linear layers and convolutions,
+are much faster in ``float16``. Other operations, like reductions, often require the dynamic
+range of ``float32``. Networks running in mixed precision try to match each operation to its appropriate datatype.
+
+.. contents:: :local:
+
+.. _gradient-scaling:
+
+Gradient Scaling
+^^^^^^^^^^^^^^^^
+
+When training a network with mixed precision, gradient magnitudes may become too small to represent
+in ``torch.float16`` regions of the backward pass.  "Gradient scaling" multiplies outputs by
```

Comment:
"region of the backward pass" is not a thing. Need to be clearer about how the dtype of ops performed in forward relate to backward pass.

Reply:
Just make it really clear for a single op.

Reply:
"outputs" - of what? Also, that's what it does technically, not what does it conceptually.